libs/librepcb/common/units/length.cpp

```diff
@@ -37,10 +37,51 @@ namespace librepcb {
  ******************************************************************************/
 
 QString Length::toMmString() const noexcept {
-  QString str = QLocale::c().toString(toMm(), 'f', 6);
-  for (int i = 0; (i < 5) && str.endsWith(QLatin1Char('0')); ++i) {
-    str.chop(1);
+  using LengthBaseU_t = std::make_unsigned<LengthBase_t>::type;
+
+  LengthBase_t nm = toNm();
+  LengthBaseU_t nm_abs;
+  if (nm < 0) {
+    nm_abs = -static_cast<LengthBaseU_t>(nm);
```

Comment:
Hmm this looks strange. The typecast `static_cast<LengthBaseU_t>(nm)` evaluates to a (large) unsigned number. The minus sign then converts it back to a signed number. And the assignment to `nm_abs` leads in an implicit cast back to an unsigned integer. Or am I misunderstanding something?

Shouldn't it just be that way?

```suggestion
    nm_abs = static_cast<LengthBaseU_t>(-nm);
```

Reply:
AKAIC, signed overflow has undefined behavior, whereas unsigned overflow is well defined. The negation operator on a signed integer can overflow when the operand is the minimum value. The negation operator on an unsigned integer (bigger than `short`) will compute the two's complement negation and return it as an unsigned integer.

That's why I did the negation on the unsigned number, to handle the situation where `nm` has the minimum value, even though it's a corner case.

Reply:
Ah, good point! :smiley: